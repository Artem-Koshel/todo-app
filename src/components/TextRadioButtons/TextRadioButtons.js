import React, { useCallback } from 'react';
import { PropTypes } from 'prop-types';
import styles from './TextRadioButtons.module.scss';

export default function TextRadioButtons({ name, selectedValue, values, onChange }) {
  const handleValueChange = useCallback((event) => {
    onChange(event.target.value);
  });

  const textButtons = values.map((value) => {
    const selected = value === selectedValue;
    return (
      <label key={value} htmlFor={value}>
        <input
          type="radio"
          id={value}
          name={name}
          value={value}
          onChange={handleValueChange}
          checked={selected}
        />
        <span className={selected ? styles.checked : undefined}>{value}</span>
      </label>
    );
  });

  return <div className={styles.textRadioButtons}>{textButtons}</div>;
}

TextRadioButtons.propTypes = {
  name: PropTypes.string.isRequired,
  selectedValue: PropTypes.string.isRequired,
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
};
