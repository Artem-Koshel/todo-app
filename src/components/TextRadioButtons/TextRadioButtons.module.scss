.textRadioButtons {
    label {
        margin-left: 10px;
        cursor: pointer;
        font-weight: bold;
    }
    
    input {
        display: none;
    }

    .checked {
        color: #968ff2;
    }
}