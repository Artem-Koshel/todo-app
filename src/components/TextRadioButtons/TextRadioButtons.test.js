import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TextRadioButtons from './TextRadioButtons';

describe('TextRadioButtons component', () => {
  it('renders correctly', () => {
    const values = ['value1', 'value2', 'value3'];
    const { container } = render(
      <TextRadioButtons
        name="test name"
        selectedValue="value2"
        values={values}
        onChange={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('exec callback on change', async () => {
    const user = userEvent.setup();
    const callback = jest.fn();
    const { container } = render(
      <TextRadioButtons
        name="test name"
        selectedValue="value2"
        values={['value1', 'value2', 'value3']}
        onChange={callback}
      />
    );

    await user.click(container.querySelector('#value3'));

    expect(callback).toBeCalledWith('value3');
  });
});
