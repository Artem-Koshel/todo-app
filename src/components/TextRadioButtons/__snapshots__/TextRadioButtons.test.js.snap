// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextRadioButtons component renders correctly 1`] = `
<div>
  <div
    class="textRadioButtons"
  >
    <label
      for="value1"
    >
      <input
        id="value1"
        name="test name"
        type="radio"
        value="value1"
      />
      <span>
        value1
      </span>
    </label>
    <label
      for="value2"
    >
      <input
        checked=""
        id="value2"
        name="test name"
        type="radio"
        value="value2"
      />
      <span
        class="checked"
      >
        value2
      </span>
    </label>
    <label
      for="value3"
    >
      <input
        id="value3"
        name="test name"
        type="radio"
        value="value3"
      />
      <span>
        value3
      </span>
    </label>
  </div>
</div>
`;
